app/controllers/application_controller.rb

```diff
@@ -14,6 +15,16 @@ def root
 
   private
 
+  def authenticate
+    sign_in(model_by_param(User, :email))
+  rescue
+    'Model cannot be authenticate'
+  end
+
+  def model_by_param(model, param)
+    model.find_by("#{param}": params[model.name.underscore][param])
```

Comment:
no!
param this is variable from function argument
```
def model_by_param(model, **param**)
```

Such an implementation allows this `"#{param}"` to be an argument for the find_by function
example: Post.find_by(user_id: user.id)